java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java

```diff
@@ -642,9 +660,38 @@ private static void alignEnvironment(ContentEnvironment env, Channel baseChannel
                 .sorted((t1, t2) -> Boolean.compare(t1.getChannel().isBaseChannel(), t2.getChannel().isBaseChannel()))
                 .forEach(toRemove -> ContentProjectFactory.purgeTarget(toRemove));
 
-        // align the contents
-        newSrcTgtPairs.forEach(srcTgt ->
-                alignEnvironmentTarget(srcTgt.getLeft(), srcTgt.getRight(), filters, async, user));
+        // Strip modules metadata from the modular repositories if any modular filter is present
+        if (filters.stream().anyMatch(f -> f.asModuleFilter().isPresent())) {
+            newSrcTgtPairs.stream().map(p -> p.getRight().getChannel()).forEach(this::stripModuleMetadata);
+        }
+
+        // Resolve filters for dependencies
+        try {
+            DependencyResolver resolver = new DependencyResolver(env.getContentProject(), this.modulemdApi);
+            List<ContentFilter> resolvedFilters = resolver.resolveFilters(filters);
+
+            // align the contents
+            newSrcTgtPairs.forEach(srcTgt ->
+                    alignEnvironmentTarget(srcTgt.getLeft(), srcTgt.getRight(), resolvedFilters, async, user));
+        }
+        catch (DependencyResolutionException e) {
+            if (e.getCause() instanceof ModuleNotFoundException) {
+                ModuleNotFoundException cause = (ModuleNotFoundException) e.getCause();
+                LOG.debug(String.format("Module '%s:%s' not found.", cause.getModule().getName(),
```

Comment:
Is `DEBUG` enough here? I think the log threshold is set to `INFO` in regular setups - i.e. this message would be silenced. With my current understanding of the PR, this line shouldn't get hit on a correctly set-up project, so I'd be in favor of using at least `INFO` here.

Reply:
A more sophisticated error handling is also one of the next steps for the epic (https://github.com/SUSE/spacewalk/issues/10099), but I agree, I'm changing to `INFO` for now.